### PR TITLE
More warnings

### PR DESCRIPTION
**Pull request by jkbaumohl**

### CODE REVIEW

File: lib/GenomeFileUtil/core/FastaGFFToGenome.py [hunk 1]
```diff
@@ -805,6 +855,47 @@ def _gen_genome_info(self, core_genome_name, scientific_name, assembly_ref,
...
+            is_transpliced = False
+            if "flags" in feature and "trans_splicing" in feature["flags"]:
+                is_transpliced = True      
+            if not is_transpliced and len(feature["location"]) > 1:
+                # Check the order only if not trans_spliced and has more than 1 location.    
+                location_warning = self._check_location_order(feature["location"])
+                if location_warning is not None:
+                    feature["warnings"] = feature.get('warnings', []) + [location_warning]     
+
+            # Test for full contig length features and if on both strands.
```
[the remark this answers is not in the document]
Comment (JamesJeffryes) [reply lib/GenomeFileUtil/core/FastaGFFToGenome.py]: This helper function should be in GenomeUtils and use the helper functions already there

File: lib/GenomeFileUtil/core/FastaGFFToGenome.py [hunk 2]
```diff
@@ -805,6 +855,47 @@ def _gen_genome_info(self, core_genome_name, scientific_name, assembly_ref,
...
+            for location in feature["location"]:
+                location_contigs.add(location[0])
+            if len(location_contigs) == 1:
+                for location in feature["location"]:
+                    if location[2] == "+":
+                        strand_set.add("+")
+                        location_min = location[1]  
+                        location_max = location[1] + location[3] - 1
+                    if location[2] == "-":
+                        strand_set.add("+")
```
Comment (JamesJeffryes) [inline lib/GenomeFileUtil/core/FastaGFFToGenome.py]: I think you mean "-"

File: lib/GenomeFileUtil/core/FastaGFFToGenome.py [hunk 3]
```diff
@@ -805,6 +855,47 @@ def _gen_genome_info(self, core_genome_name, scientific_name, assembly_ref,
...
+                        location_min = location[1] + location[3] + 1   
+                        location_max = location[1]                       
+                    if feature_min_location is None or feature_min_location > location_min:
+                        feature_min_location = location_min
+                    if feature_max_location is None or feature_max_location < location_max:
+                        feature_max_location = location_max
+                if feature_min_location == 1 \
```
Comment (JamesJeffryes) [inline lib/GenomeFileUtil/core/FastaGFFToGenome.py]: Why not just check the length of the first location in the feature to see if it's equal to contig length

File: lib/GenomeFileUtil/core/FastaGFFToGenome.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamesJeffryes) [reply lib/GenomeFileUtil/core/FastaGFFToGenome.py]: Scratch this, see my later comment for a way to make this shorter

File: lib/GenomeFileUtil/core/FastaGFFToGenome.py [hunk 4]
```diff
@@ -805,6 +855,47 @@ def _gen_genome_info(self, core_genome_name, scientific_name, assembly_ref,
                 if 'exon' in feature:
                     self._update_from_exons(feature)
                 genome['non_coding_features'].append(feature)
+
+            # Test if location order is in order.
+            is_transpliced = False
+            if "flags" in feature and "trans_splicing" in feature["flags"]:
+                is_transpliced = True      
+            if not is_transpliced and len(feature["location"]) > 1:
```
[the remark this answers is not in the document]
Comment (jkbaumohl) [reply lib/GenomeFileUtil/core/FastaGFFToGenome.py]: I use the "is_transpliced" flag below. Keeping this as is.

File: lib/GenomeFileUtil/core/FastaGFFToGenome.py (hunk 2 above)
Comment (jkbaumohl) [reply lib/GenomeFileUtil/core/FastaGFFToGenome.py]: Yeah you are right. Confused now on why it threw the warning. If I add "+", then "+" to a set, it only contains 1 "+" correct?

File: lib/GenomeFileUtil/core/FastaGFFToGenome.py (hunk 3 above)
Comment (jkbaumohl) [reply lib/GenomeFileUtil/core/FastaGFFToGenome.py]: What if it is a multi_location feature?

**Pull request merged**